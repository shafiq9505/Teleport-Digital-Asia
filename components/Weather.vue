<template>
	<!-- Display Weather Data-->
	<b-row>
		<b-col class="mt-3" v-show="fields.weatherData.fieldAttr.show">
			<b-card class="mb-lg" no-body>
				<b-card-header>
					<strong>Weather in {{ weatherObj.name }}</strong>
				</b-card-header>
				<b-card-body class="pl-2">
					<b-container>
						<!-- Weather Details -->
						<b-row>
							<!-- City -->
							<b-col cols="12">
								<b-form-group horizontal :label-cols="4" label="City">
									<b-form-input readonly v-model="weatherObj.name"></b-form-input>
								</b-form-group>
							</b-col>
						</b-row>
						<b-row>
							<!-- Weather -->
							<b-col class="mt-3" cols="12">
								<b-form-group horizontal :label-cols="4" label="Weather">
									<b-form-input readonly v-model="weatherObj.weather[0].description"></b-form-input>
								</b-form-group>
							</b-col>

							<!-- Humidity -->
							<b-col class="mt-3" cols="12">
								<b-form-group horizontal :label-cols="4" label="Humidty">
									<b-form-input readonly v-model="weatherObj.main.humidity"></b-form-input>
								</b-form-group>
							</b-col>

							<!-- Pressure -->
							<b-col class="mt-3" cols="12">
								<b-form-group horizontal :label-cols="4" label="Pressure">
									<b-form-input readonly v-model="weatherObj.main.pressure"></b-form-input>
								</b-form-group>
							</b-col>

							<!--Temperature  -->
							<b-col class="mt-3" cols="12">
								<b-form-group horizontal :label-cols="4" label="Temperature">
									<b-form-input readonly v-model="weatherObj.main.temp"></b-form-input>
								</b-form-group>
							</b-col>

							<!-- Max Temperature -->
							<b-col class="mt-3" cols="12">
								<b-form-group horizontal :label-cols="4" label="Max Temperature">
									<b-form-input readonly v-model="weatherObj.main.temp_max"></b-form-input>
								</b-form-group>
							</b-col>

							<!-- Min Temperature -->
							<b-col class="mt-3" cols="12">
								<b-form-group horizontal :label-cols="4" label="Min Temperature">
									<b-form-input readonly v-model="weatherObj.main.temp_min"></b-form-input>
								</b-form-group>
							</b-col>
						</b-row>

						<!-- Button -->
						<b-row v-show="!hideGetLocationButton" class="mt-2">
							<b-col cols="12">
								<b-btn class="float-right ml-1 mt-0" variant="primary" @click="getCurrentLocation()">Get current location</b-btn>
							</b-col>
						</b-row>
					</b-container>
				</b-card-body>
			</b-card>
		</b-col>
	</b-row>
</template>

<script>
export default {
	name: 'Weather',
	data() {
		return {
			formData: {
				weather: {
					sky: '',
					shumidityky: ''
				}
			},
			fields: {
				weatherData: {
					options: [],
					fieldAttr: {
						show: true
					}
				}
			}
		}
	},

	props: {
		weatherObj: {
			type: Object,
			required: true
		},
		hideGetLocationButton: {
			type: Boolean,
			required: false,
			default: false
		}
	},
	async mounted() {},
	methods: {
		getCurrentLocation() {
			this.$emit('getCurrentLocation')
		}
	}
}
</script>