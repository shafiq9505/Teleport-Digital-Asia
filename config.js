module.exports = {
	APP_VERSION: 'x.y.z-sss',
    WEATHER_API_KEY : '283d16e34cf45af9d977478080dae918',
	URL_PATHS: {
		WEATHER: {
			LOGOUT: 'api.openweathermap.org/data/2.5/weather'
		},
	
	},

}
