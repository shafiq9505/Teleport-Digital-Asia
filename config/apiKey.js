export const apiKey = '283d16e34cf45af9d977478080dae918'