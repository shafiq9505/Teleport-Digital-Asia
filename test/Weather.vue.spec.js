import Weather from '@/components/Weather.vue'
import { shallowMount, } from '@vue/test-utils'
import {stubs} from './config/bootStrapVue'
import _ from 'lodash'

// create an extended `Vue` constructor



let wrapper 
beforeEach(() => {
    
	wrapper = shallowMount(Weather, {
		propsData: {
            weatherObj : {
                name : 'Kuala Lumpur',
                main : {
                    description : 'testing',
                    humidity : '123',
                    pressure : '123',
                    temp : '123',
                    temp_max : '123',
                    temp_min : '123'
                },
                
                weather : [{
                    description : ''
                }]
            }
        },
        stubs,
	})
})

describe('data', () => {
	it('default assigned data', () => {
		expect(Weather.data()).toMatchSnapshot()
	})
})



describe('methods', () => {
	describe('getCurrentLocation', () => {
        it('emit current location', () => {
            wrapper.vm.$emit = jest.fn()
            wrapper.vm.getCurrentLocation()
            expect(wrapper.vm.$emit).toBeCalledWith('getCurrentLocation')
        })
    })
})
