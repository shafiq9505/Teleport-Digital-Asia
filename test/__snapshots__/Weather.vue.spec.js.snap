// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data default assigned data 1`] = `
Object {
  "fields": Object {
    "weatherData": Object {
      "fieldAttr": Object {
        "show": true,
      },
      "options": Array [],
    },
  },
  "formData": Object {
    "weather": Object {
      "shumidityky": "",
      "sky": "",
    },
  },
}
`;

exports[`methods getCurrentLocation emit current location 1`] = `
Object {
  "fields": Object {
    "weatherData": Object {
      "fieldAttr": Object {
        "show": true,
      },
      "options": Array [],
    },
  },
  "formData": Object {
    "weather": Object {
      "shumidityky": "",
      "sky": "",
    },
  },
}
`;
